polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -444,26 +449,89 @@ impl<Config: config::Config> XcmExecutor<Config> {
 		Ok(())
 	}
 
-	fn take_fee(&mut self, fee: Assets, reason: FeeReason) -> XcmResult {
+	fn take_fee(&mut self, fees: Assets, reason: FeeReason) -> XcmResult {
 		if Config::FeeManager::is_waived(self.origin_ref(), reason.clone()) {
 			return Ok(())
 		}
 		log::trace!(
 			target: "xcm::fees",
-			"taking fee: {:?} from origin_ref: {:?} in fees_mode: {:?} for a reason: {:?}",
-			fee,
+			"taking fees: {:?} from origin_ref: {:?} in fees_mode: {:?} for a reason: {:?}",
+			fees,
 			self.origin_ref(),
 			self.fees_mode,
 			reason,
 		);
-		let paid = if self.fees_mode.jit_withdraw {
-			let origin = self.origin_ref().ok_or(XcmError::BadOrigin)?;
-			for asset in fee.inner() {
-				Config::AssetTransactor::withdraw_asset(&asset, origin, Some(&self.context))?;
+		// We only ever use the first asset from `fees`.
+		let asset_needed_for_fees = match fees.get(0) {
```

Comment:
how can we make sure no one requires multiple assets as a fee payment? if anyone does, this silently will break things.

Reply:
This was discussed publicly on the [XCMv5 RFC targeting new way to specify fees](https://github.com/polkadot-fellows/xcm-format/pull/53).

Consensus there was that in practice no chain will ask fees of multiple asset types at once. It'd just be bad design. They might _support_ multiple assets for paying fees, in which case any one of the supported assets can be used to cover all required fees.